.github/actions/setup-aws-cli/action.yml

```diff
@@ -0,0 +1,16 @@
+name: 'Setup awscli'
+description: 'Set up a specific version of awscli'
+inputs:
+  version:
+    description: 'Version of awscli'
+    required: false
+    default: 2.4.25
+runs:
+  using: "composite"
+  steps:
+    - run: | 
+        AWSCLI_VERSION=${{ inputs.version }}
```

Comment:
prefer using `env` section to provide variables for a run

here you don't really to set a variable, just use ${{ inputs.version }} in curl command